miio/tests/test_ceil.py

```
@@ -0,0 +1,126 @@
+            'apply_fixed_scene': lambda x: self._set_state("snm", x),
+            'delay_off': lambda x: self._set_state("dv", x),
+            'enable_bl': lambda x: self._set_state("bl", x),
+            'enable_ac': lambda x: self._set_state("ac", x),
+            'set_cct': lambda x: self._set_state("cct", x),
+        }
+        super().__init__(args, kwargs)
+
+
+@pytest.fixture(scope="class")
+def ceil(request):
+    request.cls.device = DummyCeil()
+    # TODO add ability to test on a real device
+
+
+@pytest.mark.usefixtures("ceil")
+class TestCeil(TestCase):
+    def is_on(self):
+        return self.device.status().is_on
+
+    def state(self):
+        return self.device.status()
+
+    def test_on(self):
+        self.device.off()  # ensure off
+
+        start_state = self.is_on()
+        assert start_state is False
+
+        self.device.on()
+        assert self.is_on() is True
+
+    def test_off(self):
+        self.device.on()  # ensure on
+
+        assert self.is_on() is True
+        self.device.off()
+        assert self.is_on() is False
+
+    def test_status(self):
+        self.device._reset_state()
+
+        assert self.is_on() is True
+        assert self.state().brightness == self.device.start_state["bright"]
+        assert self.state().color_temperature == self.device.start_state["cct"]
+        assert self.state().scene == self.device.start_state["snm"]
+        assert self.state().delay_off_countdown == self.device.start_state["dv"]
+        assert self.state().smart_night_light == (self.device.start_state["bl"] == 1)
+        assert self.state().automatic_color_temperature == (self.device.start_state["ac"] == 1)
+
+    def test_set_brightness(self):
+        def brightness():
+            return self.device.status().brightness
+
+        self.device.set_brightness(10)
+        assert brightness() == 10
+        self.device.set_brightness(20)
+        assert brightness() == 20
```

Test against incorrect values (brightness <0, >100), the bulbs should raise an exception if an incorrect value is thrown at them I think.